File: app/src/main/java/space/pal/sig/model/database/converters/MediaTypeConverter.java
package space.pal.sig.model.database.converters;

import androidx.room.TypeConverter;

import space.pal.sig.model.MediaType;

/**
 * SpaceNow
 * Created by Catalin on 7/24/2020
 **/
public class MediaTypeConverter {

    @TypeConverter
    public static MediaType fromString(String type) {
        return type == null ? null : MediaType.valueOf(type);
    }

    @TypeConverter
    public static String toString(MediaType type) {
        return type == null ? null : type.name();
    }

}

File: app/src/main/java/space/pal/sig/database/converters/MediaTypeConverter.kt
package space.pal.sig.database.converters

import androidx.room.TypeConverter
import space.pal.sig.old.model.MediaType

/**
 * SpaceNow
 * Created by Catalin on 7/24/2020
 */
object MediaTypeConverter {
    @TypeConverter
    fun fromString(type: String?): MediaType? {
        return if (type == null) null else MediaType.valueOf(type)
    }

    @TypeConverter
    fun toString(type: MediaType?): String? {
        return type?.name
    }
}